packages/vite/src/node/build.ts

```diff
@@ -502,22 +502,36 @@ async function doBuild(
       }
 
       const watcherOptions = config.build.watch
+      const resolvedChokidarOptions = {
+        ignoreInitial: true,
+        ignorePermissionErrors: true,
+        ...watcherOptions.chokidar,
+        ignored: [
+          '**/node_modules/**',
+          '**/.git/**',
+          ...(watcherOptions?.chokidar?.ignored || [])
+        ]
+      }
```

Comment:
I think it would be nice to extract this as a function so it can be used by `server/index.ts` too. We could also have the warning check located in it, even though both are slightly different but I think it'd be helpful to have both near in-code.

Looks like in this code, we should be ignoring `'**/test-results/**'` too.

Reply:
I've extracted into a function 👍 : 0eda07177b87db379b1366c7f9a5273e927789b1